test/unidirectional-streams.spec.js

```diff
@@ -1,180 +1,108 @@
-    ]
-
-    // server context - waits for the client to connect, opens a bidi stream, sends some data and reads the response
-    Promise.resolve().then(async () => {
-      const session = await getReaderValue(server.sessionStream(SERVER_PATH))
-      const stream = await session.createUnidirectionalStream()
-
-      await writeStream(stream, input)
-    })
-
     // client context - waits for the server to open a bidi stream then pipes it back to them
-    client = new WebTransport(`${url}${SERVER_PATH}`, {
-      serverCertificateHashes: [
-        {
-          algorithm: 'sha-256',
-          value: certificate.hash
-        }
-      ]
-    })
+    client = new WebTransport(
+      `${process.env.SERVER_URL}/unidirectional_server_send`,
+      {
+        serverCertificateHashes: [
+          {
+            algorithm: 'sha-256',
+            value: readCertHash(process.env.CERT_HASH)
+          }
+        ]
+      }
+    )
     await client.ready
 
     const stream = await getReaderValue(client.incomingUnidirectionalStreams)
-    const received = await readStream(stream)
-    expect(received).to.deep.equal(
-      input,
+    const output = await readStream(stream, KNOWN_BYTES.length)
+    expect(ui8.concat(KNOWN_BYTES)).to.deep.equal(
+      ui8.concat(output),
       'Did not receive the same bytes we sent'
     )
   })
 
   it('handles fin when paused due to backpressure', async function () {
-    this.timeout(10000)
-
-    /** @type {Deferred<ReadableStream>} */
-    const serverStream = defer()
-
-    Promise.resolve().then(async () => {
-      const session = await getReaderValue(server.sessionStream(SERVER_PATH))
-      const stream = await getReaderValue(session.incomingUnidirectionalStreams)
-
-      serverStream.resolve(stream)
-    })
-
-    client = new WebTransport(`${url}${SERVER_PATH}`, {
-      serverCertificateHashes: [
-        {
-          algorithm: 'sha-256',
-          value: certificate.hash
-        }
-      ]
-    })
+    this.timeout(5000)
+
+    client = new WebTransport(
+      `${process.env.SERVER_URL}/unidirectional_server_delay_before_read`,
+      {
+        serverCertificateHashes: [
+          {
+            algorithm: 'sha-256',
+            value: readCertHash(process.env.CERT_HASH)
+          }
+        ]
+      }
+    )
     await client.ready
 
     const clientStream = await client.createUnidirectionalStream()
 
-    const input = [
-      Uint8Array.from([0, 1, 2, 3, 4]),
-      Uint8Array.from([5, 6, 7, 8, 9]),
-      Uint8Array.from([10, 11, 12, 13, 14]),
-      Uint8Array.from([15, 16, 17, 18, 19]),
-      Uint8Array.from([20, 21, 22, 23, 24])
-    ]
-
     const writer = clientStream.getWriter()
 
-    for (const buf of input) {
+    for (const buf of KNOWN_BYTES) {
+      await writer.ready
       await writer.write(buf)
     }
 
-    await new Promise((resolve) => setTimeout(resolve, 1000))
+    await new Promise((resolve) => setTimeout(resolve, 500))
```

Comment:
Increasing the timeout increases the likely hood of a Connection Lost or Session closed